components/core/src/package/metadata.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::str::FromStr;
+
+use error::{Error, Result};
+
+#[derive(Debug)]
+pub struct Bind {
+    pub service: String,
+    pub exports: Vec<String>,
+}
+
+impl FromStr for Bind {
+    type Err = Error;
+
+    fn from_str(line: &str) -> Result<Self> {
+        let mut parts = line.split('=');
+        let service = match parts.next() {
+            None => return Err(Error::MetaFileBadBind),
+            Some(service) => service.to_string(),
+        };
+        let exports = match parts.next() {
```

Comment:
This is pretty cool use of the iterator and `next()` to line up either side of the key/value from a string for their respective `match`.